src/re.rs

```diff
@@ -970,11 +970,11 @@ impl<'t> Index<usize> for Captures<'t> {
 ///
 /// # Panics
 /// If there is no group named by the given value.
-impl<'t> Index<&'t str> for Captures<'t> {
+impl<'t, 'i> Index<&'i str> for Captures<'t> {
```

Comment:
Could you please add a bit of documentation here that briefly describes what the `'i` lifetime parameter stands for? Thanks! (The `'t` lifetime should also be documented, e.g., http://doc.rust-lang.org/regex/regex/struct.FindMatches.html)